File: app/src/main/java/nl/rsdt/japp/jotial/data/bodies/HunterPostBody.java
package nl.rsdt.japp.jotial.data.bodies;

import com.google.android.gms.maps.model.LatLng;
import com.google.gson.JsonObject;
import com.google.gson.annotations.SerializedName;

import org.json.JSONObject;

import java.util.HashMap;

import nl.rsdt.japp.application.JappPreferences;

/**
 * @author Dingenis Sieger Sinke
 * @version 1.0
 * @since 14-8-2016
 * Description...
 */
public class HunterPostBody {

    @SerializedName("SLEUTEL")
    private String sleutel;

    @SerializedName("hunter")
    private String hunter;

    @SerializedName("latitude")
    private String latitude;

    @SerializedName("longitude")
    private String longitude;

    @SerializedName("icon")
    private String icon;

    public HunterPostBody setKey(String key) {
        this.sleutel = key;
        return this;
    }

    public HunterPostBody setName(String name)
    {
        this.hunter = name;
        return this;
    }

    public HunterPostBody setLatLng(LatLng latLng)
    {
        this.latitude = Double.toString(latLng.latitude);
        this.longitude = Double.toString(latLng.longitude);
        return this;
    }

    public HunterPostBody setIcon(int icon) {
        this.icon = Integer.toString(icon);
        return this;
    }

    public static HunterPostBody getDefault() {
        HunterPostBody builder = new HunterPostBody();

        String huntname = JappPreferences.getHuntname();
        if(!huntname.isEmpty()) {
            builder.setName(huntname);
        } else {
            builder.setName(JappPreferences.getAccountUsername());
        }

        builder.setKey(JappPreferences.getAccountKey());
        builder.setIcon(JappPreferences.getAccountIcon());
        return builder;
    }




}

File: app/src/main/java/nl/rsdt/japp/jotial/data/bodies/HunterPostBody.kt
package nl.rsdt.japp.jotial.data.bodies

import com.google.android.gms.maps.model.LatLng
import com.google.gson.annotations.SerializedName

import nl.rsdt.japp.application.JappPreferences
import nl.rsdt.japp.jotial.maps.deelgebied.Deelgebied

/**
 * @author Dingenis Sieger Sinke
 * @version 1.0
 * @since 14-8-2016
 * Description...
 */
class HunterPostBody {

    @SerializedName("SLEUTEL")
    private var sleutel: String? = null

    @SerializedName("hunter")
    private var hunter: String? = null

    @SerializedName("latitude")
    private var latitude: String? = null

    @SerializedName("longitude")
    private var longitude: String? = null

    @SerializedName("icon")
    private var icon: String? = null

    fun setKey(key: String?): HunterPostBody {
        this.sleutel = key
        return this
    }

    fun setName(name: String?): HunterPostBody {
        this.hunter = name
        return this
    }

    fun setLatLng(latLng: LatLng): HunterPostBody {
        this.latitude = latLng.latitude.toString()
        this.longitude = latLng.longitude.toString()
        return this
    }

    fun setIcon(icon: Int): HunterPostBody {
        this.icon = icon.toString()
        return this
    }

    fun prependDeelgebiedToName(dg: Deelgebied): HunterPostBody{
        hunter = hunter?.replace('.', ' ')
        if (JappPreferences.prependDeelgebied){
            hunter = "${dg.name}.$hunter"
        }  else if(icon == "0"){
            icon = "1"
        }
        return this
    }
    companion object {

        val default: HunterPostBody
            get() {
                val builder = HunterPostBody()
                var icon = JappPreferences.accountIcon
                val dg = JappPreferences.taak
                var huntname = JappPreferences.huntname
                if (huntname.isEmpty()){
                    huntname = JappPreferences.accountUsername
                }

                builder.setKey(JappPreferences.accountKey)
                builder.setIcon(icon)
                builder.setName(huntname)
                return builder
            }
    }


}
